src/fiatconnect/ReviewScreen.tsx

```diff
@@ -0,0 +1,340 @@
+      <View>
+        <Amount flow={flow} cicoQuote={cicoQuote} tokenAddress={tokenInfo.address} />
+        <TransactionDetails flow={flow} cicoQuote={cicoQuote} tokenAddress={tokenInfo.address} />
+        <PaymentMethod
+          provider={provider}
+          fiatAccount={fiatAccount}
+          fiatAccountSchema={fiatAccountSchema}
+          fiatAccountLogo={fiatAccountLogo}
+        />
+      </View>
+      <Button
+        style={styles.submitBtn}
+        type={BtnTypes.PRIMARY}
+        size={BtnSizes.FULL}
+        text={
+          flow === CICOFlow.CashIn
+            ? t('fiatConnectReviewScreen.cashIn.button')
+            : t('fiatConnectReviewScreen.cashOut.button')
+        }
+        onPress={() => {
+          Logger.debug(TAG, 'Button clicked')
+
+          // TODO(any): submit the transfer
+        }}
+      />
+    </SafeAreaView>
+  )
+}
+
+function Amount({
+  flow,
+  cicoQuote,
+  tokenAddress,
+}: {
+  flow: CICOFlow
+  cicoQuote: QuoteResponse
+  tokenAddress: string
+}) {
+  return (
+    <View style={styles.amountContainer}>
+      {flow === CICOFlow.CashIn ? (
+        <TokenDisplay
+          amount={cicoQuote.quote.cryptoAmount}
+          tokenAddress={tokenAddress}
+          style={styles.amountText}
+          showLocalAmount={false}
+          testID="amount-crypto"
+        />
+      ) : (
+        <CurrencyDisplay
+          style={styles.amountText}
+          amount={{
+            value: cicoQuote.quote.cryptoAmount,
+            currencyCode: cicoQuote.quote.cryptoType,
+            localAmount: {
+              value: cicoQuote.quote.fiatAmount,
+              currencyCode: cicoQuote.quote.fiatType,
+              exchangeRate: 1,
+            },
+          }}
+          testID="amount-fiat"
+        />
+      )}
+    </View>
+  )
+}
+
+function TransactionDetails({
+  flow,
+  cicoQuote,
+  tokenAddress,
+}: {
+  flow: CICOFlow
+  cicoQuote: QuoteResponse
+  tokenAddress: string
+}) {
+  const { t } = useTranslation()
+  let tokenDisplay: string
+  switch (cicoQuote.quote.cryptoType) {
+    case CryptoType.cUSD:
+      tokenDisplay = t('celoDollar')
+      break
+    case CryptoType.cEUR:
+      tokenDisplay = t('celoEuro')
+      break
+    default:
+      tokenDisplay = t('total')
+  }
+  return (
+    <View style={styles.sectionContainer}>
+      <Text style={styles.sectionHeaderText}>
+        {t('fiatConnectReviewScreen.transactionDetails')}
+      </Text>
+      <LineItemRow
+        style={styles.sectionMainTextContainer}
+        textStyle={styles.sectionMainText}
+        title={
+          flow === CICOFlow.CashIn
+            ? t('fiatConnectReviewScreen.cashIn.transactionDetailsAmount')
+            : t('fiatConnectReviewScreen.cashOut.transactionDetailsAmount')
+        }
+        amount={
+          <CurrencyDisplay
+            amount={{
+              value: cicoQuote.quote.cryptoAmount,
+              currencyCode: cicoQuote.quote.cryptoType,
```

Comment:
CurrencyDisplay component uses `useLocalCurrencyToShow` hook to convert amount to local currency amount. it assumes `currencyCode` to be Type of `Currency`. `CryptoType` enum is not exactly the same as `Currency` type, specifically, `Currency` calls `Celo` `cGLD` instead of `CELO` and does not include cREAL.

Reply:
these values actually don't matter, see https://github.com/valora-inc/wallet/pull/2504#discussion_r882273431